client/src/modules/list/components/Items/ListItemDescription/ListItemDescription.jsx

```diff
@@ -0,0 +1,275 @@
+} from 'common/components/Preloader';
+import { RouterMatchPropType } from 'common/constants/propTypes';
+import { updateListItem } from '../model/actions';
+import { KeyCodes } from 'common/constants/enums';
+import { AbortPromiseException } from 'common/exceptions/AbortPromiseException';
+import { makeAbortablePromise } from 'common/utils/helpers';
+
+class ListItemDescription extends PureComponent {
+  pendingPromise = null;
+
+  constructor(props) {
+    super(props);
+
+    const { description } = this.props;
+    const trimmedDescription = description.trim();
+
+    this.state = {
+      isDescriptionUpdated: false,
+      descriptionInputValue: trimmedDescription,
+      isTextareaVisible: false,
+      isFocused: false,
+      pending: false
+    };
+
+    this.descriptionTextarea = React.createRef();
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    this.isDescriptionUpdated();
+    const { isTextareaVisible: previousIsTextareaVisible } = prevState;
+    const { isTextareaVisible } = this.state;
+
+    if (!previousIsTextareaVisible && isTextareaVisible) {
+      this.descriptionTextarea.current.focus();
+      this.handleFocus();
+    }
+
+    if (previousIsTextareaVisible && !isTextareaVisible) {
+      this.handleBlur();
+    }
+  }
+
+  componentWillUnmount() {
+    if (this.pendingPromise) {
+      this.pendingPromise.abort();
+    }
+  }
+
+  handleKeyPress = event => {
+    const { code } = event;
+    const { isFocused } = this.state;
+
+    if (isFocused && code === KeyCodes.ESCAPE) {
+      this.handleBlur();
+    }
+  };
+
+  handleFocus = () => {
+    this.setState({ isFocused: true });
+    document.addEventListener('keydown', this.handleKeyPress);
+  };
+
+  handleBlur = () => {
+    this.setState({ isFocused: false });
+    document.removeEventListener('keydown', this.handleKeyPress);
+    this.handleDescriptionUpdate();
+  };
+
+  handleShowTextarea = () => this.setState({ isTextareaVisible: true });
+
+  handleHideTextarea = () => this.setState({ isTextareaVisible: false });
+
+  handleDescriptionVisibility = () =>
+    this.setState(({ isDescriptionVisible }) => ({
+      isDescriptionVisible: !isDescriptionVisible
+    }));
+
+  handleDescriptionUpdate = () => {
+    const {
+      descriptionInputValue: description,
+      isDescriptionUpdated
+    } = this.state;
+    const {
+      disabled,
+      itemId,
+      match: {
+        params: { id: listId }
+      },
+      name,
+      updateListItem
+    } = this.props;
+
+    if (disabled) {
+      return;
+    }
+
+    if (isDescriptionUpdated) {
+      this.setState({ pending: true });
+
+      this.pendingPromise = makeAbortablePromise(
+        updateListItem(listId, itemId, { description }, name)
+      );
+      this.pendingPromise.promise
+        .then(() => {
+          this.setState({ pending: false });
+          this.handleHideTextarea();
+        })
+        .catch(err => {
+          if (!(err instanceof AbortPromiseException)) {
```

Comment:
will it ever go inside this if??

If the error of instance AbortPromiseException occurs, so the condition will be true, and there is a negation before - it will be evaluated finally to false 
the code inside will not be executed.

but if the will not be the instance of Abort Promise Exception it will go inside and set pending to false,

can't we just this.setState({pending: false}) always, independently if the error is AbortPromiseException or default error??

Reply:
Yes, it will be executed. When the component is unmounted the AbortPromiseException is thrown, then this code inside this if statement won't be executed, and it shouldn't be executed. That's why this check was created for. To prevent calling setState when the component is unmounted. In all other cases (errors) this setState will be executed.
So If a component is unmounted, AbortPromiseException is thrown, and this check prevents calling setState on unmounting component